File: go/disjoint_set/disjoint_set.go
package disjoint_set

type DisjointSet struct {
	// TODO
}

// DisjointSet constructor should initialize a disjoint set
func NewDisjointSet(n int) *DisjointSet {
	panic("TODO")
}

// Union should return false if elements i and j are already in the same set
// or perform Union operation on their sets and return true
func (set *DisjointSet) Union(i, j int) bool {
	panic("TODO")
}

// Find should return set ID: if Find(x) == Find(y), then both x and y are in the same set
func (set *DisjointSet) Find(i int) int {
	panic("TODO")
}

File: python/disjoint_set/disjoint_set.py

class DisjointSet:
	# constructor should initialize a disjoint set
	def __init__(self, n: int):
		raise NotImplementedError('TODO')

	# union should return false if elements i and j are already in the same set
	# or perform union operation on their sets and return true
	def union(self, i: int, j: int) -> bool:
		raise NotImplementedError('TODO')

	# find should return set ID: if find(x) == find(y), then both x and y are in the same set
	def find(self, i: int) -> int:
		raise NotImplementedError('TODO')
